### PR TITLE
Jorg warnings, fixes #55 and other compiler warnings

### LINKED ISSUE #55: Unused parameter warnings are annoying.
Although in most cases they are correct, they're annoying in certain functions where usage of certain params just isn't needed. There is a simple solution to squelch this: cast the parameter to `void`

To do this in every single instance is stupid though, so I propose the following:

Qt has a very nice solution: an `UNUSED` macro. This way you **explicitly** mark the variable as unused, which is also clear from the macro name. We should implement something like this ourselves. Perhaps the macro can cast the parameter to void?

### PR DESCRIPTION
**Pull request by jorgdebont**
See commit msg for more info for al the fixes

### CODE REVIEW

File: src/gamelogic/Level.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@ namespace GameLogic {
     Level::~Level()
     {
         for (auto row : this->_field) {
-            for (auto tile : row) {
+            for (Tile* tile : row) {
```
Comment (ThomasvLingen) [inline src/gamelogic/Level.cpp]: The [commit](89e41531) states that all that commit did was add a virtual destructor to Level.

Is this change needed?

Not that I really care, just curious

File: src/gamelogic/state/PauseState.cpp [hunk 2]
```diff
@@ -36,7 +37,7 @@ namespace State {
     }
 
     void PauseState::update(int time_since_last_update) {
-        // foo
```
Comment (ThomasvLingen) [inline src/gamelogic/state/PauseState.cpp]: lol, what was that foo doing there

Comment (ThomasvLingen) [inline src/gamelogic/state/PauseState.cpp]: :joy: :ok_hand:

File: src/gamelogic/Level.cpp (hunk 1 above)
Comment (jorgdebont) [inline src/gamelogic/Level.cpp]: Not sure 😂😂🐶🐶👌👌1⃣0⃣0⃣
